process/block/shardblock.go

```diff
@@ -1487,7 +1529,16 @@ func (sp *shardProcessor) addProcessedCrossMiniBlocksFromHeader(header data.Head
 				continue
 			}
 
-			sp.processedMiniBlocks.AddMiniBlockHash(string(metaBlockHash), string(miniBlockHash))
+			miniBlockHeader := process.GetMiniBlockHeaderWithHash(headerHandler, miniBlockHash)
```

Comment:
I think I have seen this in other places, maybe extract a method and reuse the common code.

Reply:
I have checked and this pattern is not common. The common code is just this method GetMiniBlockHeaderWithHash which is already extracted in common.go